### ISSUE TITLE
spatial.from_geodataframe() mangles geometries

### ISSUE THREAD

**Issue by cgarrard** [Esri/arcgis-python-api#931]
**Describe the bug**
Using `pd.DataFrame.spatial.from_geodataframe()` sometimes results in mangled geometries. It seems this is especially true when there are lots of smaller geometries surrounded by larger ones.

**To Reproduce**
```python
import pandas as pd
import geopandas as gpd
import arcgis

gdf = gpd.read_file('df_bug.gdb', layer='sample_data')
sdf = pd.DataFrame.spatial.from_geodataframe(gdf)
```

**Screenshots**
I picked an especially egregious example. This is the original data, and what it looks like after it's been read into a geodataframe, converted to a spatially enabled dataframe, and then saved to a feature class (saving to a feature class isn't necessary to reproduce the problem).

![image](https://user-images.githubusercontent.com/4453816/109356451-fabd9a00-783d-11eb-90f0-c0c7742cf4dd.png)

**Expected behavior**
I expect the geometries to stay the same.

**Platform (please complete the following information):**
 - OS: Windows 10 Pro
 - Browser: Chrome
 - Python API Version:  1.8.4

**Additional context**
The Check Geometry tool doesn't find any issues with the original data.
Sample data and notebook: [spatial_df_bug.zip](https://github.com/Esri/arcgis-python-api/files/6052645/spatial_df_bug.zip)

**Comment by achapkowski** [comment]
@AtmaMani @CMPeng please look at this issue.

**Comment by manuela-butuc** [comment]
I am having the same issue when I convert a geodataframe to a spatially enabled dataframe using GeoAccessor.from_geodataframe() function.
![image](https://user-images.githubusercontent.com/62720212/134190741-1094462c-68b3-4dbd-93bd-72d3d819c0c5.png)
becomes
![image](https://user-images.githubusercontent.com/62720212/134190830-891b9bb5-6636-4652-8ef9-f079deb7bffb.png)

**Comment by achapkowski** [comment]
This should be fixed in v2.0, please update and test.
